src/pgduckdb_filter.cpp

```diff
@@ -17,41 +19,49 @@ TemplatedFilterOperation(Datum &value, const duckdb::Value &constant) {
 	return OP::Operation((T)value, constant.GetValueUnsafe<T>());
 }
 
+template <class OP>
+bool
+StringFilterOperation(Datum &value, const duckdb::Value &constant) {
+	const auto ptr = (text*)DatumGetPointer(value);
+	if (ptr == nullptr || constant.IsNull()) {
+		return false; // Comparison to NULL always returns false.
+	}
+
+	text* tunpacked = pg_detoast_datum_packed(ptr);
+	const auto datum_sv =  std::string_view((char*)VARDATA_ANY(tunpacked), VARSIZE_ANY_EXHDR(tunpacked));
+	const auto val = duckdb::StringValue::Get(constant);
+	const auto val_sv =  std::string_view(val);
+	return OP::Operation(datum_sv, val_sv);
+}
+
 template <class OP>
 static bool
 FilterOperationSwitch(Datum &value, duckdb::Value &constant, Oid type_oid) {
 	switch (type_oid) {
 	case BOOLOID:
 		return TemplatedFilterOperation<bool, OP>(value, constant);
-		break;
 	case CHAROID:
 		return TemplatedFilterOperation<uint8_t, OP>(value, constant);
-		break;
 	case INT2OID:
 		return TemplatedFilterOperation<int16_t, OP>(value, constant);
-		break;
 	case INT4OID:
 		return TemplatedFilterOperation<int32_t, OP>(value, constant);
-		break;
 	case INT8OID:
 		return TemplatedFilterOperation<int64_t, OP>(value, constant);
-		break;
 	case FLOAT4OID:
 		return TemplatedFilterOperation<float, OP>(value, constant);
-		break;
 	case FLOAT8OID:
 		return TemplatedFilterOperation<double, OP>(value, constant);
-		break;
 	case DATEOID: {
 		Datum date_datum = static_cast<int32_t>(value + pgduckdb::PGDUCKDB_DUCK_DATE_OFFSET);
 		return TemplatedFilterOperation<int32_t, OP>(date_datum, constant);
-		break;
 	}
 	case TIMESTAMPOID: {
 		Datum timestamp_datum = static_cast<int64_t>(value + pgduckdb::PGDUCKDB_DUCK_TIMESTAMP_OFFSET);
 		return TemplatedFilterOperation<int64_t, OP>(timestamp_datum, constant);
-		break;
 	}
+	case VARCHAROID:
```

Comment:
Let's also make this work with the `text` type. Which is much more commonly used in the postgres ecosystem than `varchar`. (should have a test too)

```suggestion
	case TEXTOID:
	case VARCHAROID:
```